api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -350,6 +350,8 @@ public Role saveRole(Role role) throws APIException {
 	 */
 	@Deprecated
 	public void changePassword(User u, String pw) throws APIException {
+		OpenmrsUtil.validatePassword(u.getUsername(), pw, u.getSystemId());
```

Comment:
As i mentioned earlier, these should delegate to the changePassword method that validates the password since it does some useful checks that these need too.